mesonbuild/build.py

```diff
@@ -1616,6 +1614,15 @@ def get_clink_dynamic_linker_and_stdlibs(self) -> T.Tuple['Compiler', T.List[str
 
         raise AssertionError(f'Could not get a dynamic linker for build target {self.name!r}')
 
+    def get_used_stdlib_args(self, link_language: str) -> T.List[str]:
+        all_compilers = self.environment.coredata.compilers[self.for_machine]
+        all_langs = set(all_compilers).union(self.get_langs_used_by_deps())
```

Comment:
This seems wrong to me, please correct me if I misunderstood. The union() here is useless because all_compilers is already **all** compilers, there is nothing we can add to that set. Secondly you add stdlib link flags of all available languages to all targets regardless whether they are using that language or not. In a project that mix C++ and Fortran this will add -lc++ to a pure Fortrant target AFAIK.

Reply:
BTW, all_compilers is even all compilers from all subprojects. There is `interpreter.compilers` which is only compilers from the current subproject.

Reply:
We already called `get_langs_used_by_deps()`, instead of calling it again `dep_langs` should be passed as argument.

I believe this code was correct, the problem is not here but in `get_langs_used_by_deps()` which should recurse on dependencies.